tests/worldmap/QuadTreeTest.cpp

```diff
@@ -0,0 +1,233 @@
+	REQUIRE_FALSE(tree.remove({0, 0, 0}));
+	REQUIRE_FALSE(tree.remove({-32, 63, 1}));
+	REQUIRE_FALSE(tree.remove({-32, -84, 1}));
+	REQUIRE_FALSE(tree.remove({99, -13, 1}));
+	REQUIRE(size == tree.getSize());
+}
+
+TEST_CASE("QuadTree - GetAllPoints", "[QuadTree]")
+{
+	srand(time(nullptr)); // NOLINT(cert-msc51-cpp)
+	QuadTree tree(100, 4);
+
+	points_t points;
+
+	for (int i = 0; i < 50; i++)
+	{
+		const point_t &point = randPoint(100);
+		points.push_back(point);
+		tree.add(point);
+	}
+
+	REQUIRE(points.size() == tree.getSize());
+
+	// assert that all points are in this list
+	points_t treePoints = tree.getAllPoints();
+	REQUIRE(points.size() == treePoints.size());
+	for (const point_t &point : points)
+	{
+		auto itr = std::find(treePoints.begin(), treePoints.end(), point);
+		REQUIRE(itr != treePoints.end());
+		REQUIRE(point == *itr);
+	}
+
+	// assert that modifying this list doesn't modify the tree
+	auto prevSize = treePoints.size();
+	point_t erased = *treePoints.begin();
+	treePoints.erase(treePoints.begin());
+	// not a catch2 assert since this is just a sanity check
+	assert(prevSize == treePoints.size() + 1);
+
+	REQUIRE(tree.getAllPoints().size() == prevSize);
+	REQUIRE(tree.getClosestWithin(erased, 1) == erased);
+
+	// verify that mutating the objects in the list doesn't modify the tree
+	treePoints = tree.getAllPoints();
+	REQUIRE(tree.getClosestWithin(treePoints[0], 0.5) == treePoints[0]);
+	treePoints[0](0) = treePoints[0](0) + 0.01;
+	REQUIRE(tree.getClosestWithin(treePoints[0], 0.5) != treePoints[0]);
+}
+
+TEST_CASE("QuadTree - GetPointsWithin", "[QuadTree]")
+{
+	srand(time(nullptr)); // NOLINT(cert-msc51-cpp)
+	QuadTree tree(100, 4);
+
+	// create two bunches: bottom-left and top-right
+	points_t left, right;
+
+	for (int i = 0; i < 50; i++)
+	{
+		point_t p = randPoint(10) - point_t(20, 20, 0);
+		left.push_back(p);
+		tree.add(p);
+	}
+
+	for (int i = 0; i < 50; i++)
+	{
+		point_t p = randPoint(10) + point_t(20, 20, 0);
+		right.push_back(p);
+		tree.add(p);
+	}
+
+	REQUIRE(tree.getSize() == (left.size() + right.size()));
+
+	points_t closestToLeft = tree.getPointsWithin(point_t(-20, -20, 1), 25);
+	points_t closestToRight = tree.getPointsWithin(point_t(20, 20, 1), 25);
+
+	// verify that we retrieved all the points in the bottom-left bunch
+	REQUIRE(closestToLeft.size() == left.size());
+	for (const point_t &point : left)
+	{
+		auto itr = std::find(closestToLeft.begin(), closestToLeft.end(), point);
+		REQUIRE(itr != closestToLeft.end());
+		REQUIRE(point == *itr);
+	}
+
+	// verify that we retrieved all the points in the top-right bunch
+	REQUIRE(closestToRight.size() == right.size());
+	for (const point_t &point : right)
+	{
+		auto itr = std::find(closestToRight.begin(), closestToRight.end(), point);
+		REQUIRE(itr != closestToRight.end());
+		REQUIRE(point == *itr);
+	}
+}
+
+TEST_CASE("QuadTree - GetClosestPoint", "[QuadTree]")
+{
+	srand(time(nullptr)); // NOLINT(cert-msc51-cpp)
+	QuadTree tree(100);
```

Comment:
Oddly, I cannot find a QuadTree constructor that takes just one argument. Does this compile?

Reply:
The second one has a default value of 4, so invoking it with one parameter compiles fine I think.

https://github.com/huskyroboticsteam/PY2020/blob/ac1b87d04da18e0e709774037b455f913651b02b/src/worldmap/QuadTree.h#L25